packages/xo-server/src/xo-mixins/pool.js

```diff
@@ -0,0 +1,72 @@
+import { isEmpty } from 'lodash'
+
+export default class {
+  constructor(xo) {
+    this._xo = xo
+  }
+
+  async mergeInto({ sources, target, force }) {
+    const targetHost = this._xo.getObject(target.master)
+    const _sources = sources.map(sourceId =>
+      this._xo.getObject(sourceId, 'pool')
+    )
+    let source
+    let sourceHost
+    for (source of _sources) {
+      sourceHost = this._xo.getObject(source.master)
+      if (sourceHost.productBrand !== targetHost.productBrand) {
+        throw new Error(
+          `a ${sourceHost.productBrand} pool cannot be merged into a ${
+            targetHost.productBrand
+          } pool`
+        )
+      }
+    }
+
+    const requiredPatchesSources = {}
+    let requiredPatchesTarget = []
+    for (source of _sources) {
+      requiredPatchesTarget = requiredPatchesTarget.concat(
+        this._xo.getPatchesDifference(source.master, target.master)
+      )
+      requiredPatchesSources[source.id] = this._xo.getPatchesDifference(
+        target.master,
+        source.master
+      )
+    }
+
+    if (requiredPatchesTarget.length > 0) {
+      const targetXapi = this._xo.getXapi(target)
+      await targetXapi.installPatches({
+        patches: await targetXapi.findPatches(requiredPatchesTarget),
+      })
+    }
+
+    if (!isEmpty(requiredPatchesSources)) {
```

Comment:
`requiredPatchesSources` will never be empty, right?

Reply:
yes.